lib/moneta/pool.rb

```diff
@@ -298,18 +298,6 @@ def initialize(options = {}, &block)
     # by the manager after the ttl
     def close; end
 
-    def each_key(&block)
-      wrap(:each_key) do
-        raise NotImplementedError, "each_key is not supported on this proxy" \
-          unless supports? :each_key
-
-        return enum_for(:each_key) { adapter ? adapter.each_key.size : check_out! { adapter.each_key.size } } unless block_given?
```

Comment:
@asppsa the change in proxy forces a change here, given that the scope of the block passed in the Proxy class, is within that object, and the checkout never happens.